flink-autoscaler/src/main/java/org/apache/flink/autoscaler/topology/VertexInfo.java

```diff
@@ -22,17 +22,19 @@
 
 import lombok.Data;
 
-import java.util.Set;
+import java.util.Map;
 
 /** Job vertex information. */
 @Data
 public class VertexInfo {
 
     private final JobVertexID id;
 
-    private final Set<JobVertexID> inputs;
+    // All input vertices and the ship_strategy
+    private final Map<JobVertexID, String> inputs;
 
-    private Set<JobVertexID> outputs;
+    // All output vertices and the ship_strategy
+    private Map<JobVertexID, String> outputs;
```

Comment:
This is a bit of a hack. Fine for now but I would prefer keeping the Set with a wrapper class containing JobVertexID and a field for the ship_strategy.

Reply:
`Set<Class<JobVertexID, String>>` has a semantic difference compared to `Map<JobVertexID, String>`. It is possible that there are multiple ship_strategies with the same JobVertexID.

I don't have strong opinion for this. I'm opening for it.

Reply:
It would depend on the hashCode / equals implementation, but the same issue arrises with Map in case of duplicate JobVertexID, only that they would be silently overwritten. Let's keep it unchanged for now.